core/schema/project.graphqls

```diff
@@ -30,14 +30,17 @@ type Project {
 "A command defined in a project that can be invoked from the CLI."
 type ProjectCommand {
   "A unique identifier for this command."
-  id: String!
+  id: ProjectCommandID!
 
   "The name of the command."
   name: String!
 
   "Flags accepted by this command."
   flags: [ProjectCommandFlag!]
 
+  "The name of the type returned by this command."
+  resultType: String
```

Comment:
I went all over the place on what type to use here. I settled on String because it's simple and possible w/out major codegen updates. It also is logical in that if you were to use graphql introspection to get the type of the underlying resolver or to query for `__typename`, it would also be a string.

Looked into graphql unions+interfaces. Interfaces in particular are kind of tempting, but it's too much to handle updating each SDK w/ interface support to make sense right now.

Enums *might* make sense here too, but I wasn't sure how much of a difference it'd make or it it would be "non-graphql-y" to create an Enum for all the types that exist in the schema anyways. Open to this option still though.

Reply:
Yeah agree it's not worth it yet, but if we hit any more cases that would benefit from interface support we should definitely consider adding it. Could be pretty useful potentially.